src/test/kotlin/VehicleTest.kt

```diff
@@ -0,0 +1,112 @@
+import org.junit.Test
+import org.junit.Assert.assertEquals
+import traffic_simulation.RoadNetwork
+import traffic_simulation.Vehicle
+
+class VehicleTest {
+
+    @Test
+    fun sameVehicleDoesNotChangeAllVehiclesSize () {
+
+        Vehicle(7 , true).newVehicle( )
+        Vehicle(8 , true).newVehicle( )
+        Vehicle(9 , true).newVehicle( )
```

Comment:
Oh ja, das scheint ein Punkt zu sein. Wenn man nur den Konstruktor aufruft kann die Funktion wohl nicht auf irend einen Inhalt zugreifen.

Währen das hier erfolgreich 1 ausgibt:

    fun main(args: Array<String>) {

    val vw : Vehicle = Vehicle (10 , true)

    vw.newVehicle()

    var i : Int = 0
    for (car in vw.allVehicles){
        i = i + 1
    }
    println(i)
    }

funktioniert es hier nicht mehr - es kommt 0 raus

    fun main(args: Array<String>) {

    val vw : Vehicle = Vehicle (10 , true)

    Vehicle(10 , true).newVehicle()

    var i : Int = 0
    for (car in vw.allVehicles){
        i = i + 1
    }
    println(i)
    }

Ich mach also bei den Tests nun die Deklarationen dau